surveys/promise-rejections/survey.md

```diff
@@ -0,0 +1,152 @@
+# Node.js Promise reject use case survey
+
+Today, Node.js handles unhandled rejections by emitting a deprecation warning to stderr. The warning shows the stack where the rejection happened, and states that in future Node.js versions unhandled rejections will result on Node.js exitting with non-zero status code. We intend to remove the deprecation warning, replacing it with a stable behavior which might be different from the one described on the deprecation warning. We're running this survey to better understand how Node.js users are using Promises and how they are dealing with unhandled rejections today, so we can make an informed decision on how to move forward.
+
+## What is a Promise rejection?
+
+A Promise rejection indicates that something went wrong while executing a Promise or an async function. Rejections can occur on several situations: thorwing inside an async function or a `Promise` executor/then/catch/finally callback, when calling the `reject` callback of an `executor`, or when calling `Promise.reject`.
+
+```js
+
+Promise.reject()  // This will result in a rejection
+
+new Promise((fulfill, reject) => {
+  reject();  // This will result in a rejection
+});
+
+new Promise(() => {
+  throw new Error();  // This will result in a rejection
+}).then(() => {
+  throw new Error();  // This will result in a rejection
+}, () => {
+  throw new Error();  // This will result in a rejection
+}).catch(() => {
+  throw new Error();  // This will result in a rejection
+}).finally(() => {
+  throw new Error();  // This will result in a rejection
+});
+
+async function () {
+  throw new Error();  // This will result in a rejection
+}
+
+async function () {
+  if (a === undefined)
+    a();  // This will result in a rejection
+}
+```
+
+## What is an unhandled rejection?
+
+Rejections are designed to be handled asynchronously. In JavaScript, Promises can be handled when awaiting for a promise inside a try/catch block, or via .catch calls:
```

Comment:
I don't think that's true, rejections are _designed_ to be handled asynchronously so that errors are always in the same timing and users don't have to do `.catch` and `} catch {` both.

Promise are also generally not "handled" they are just "reacted to".

So I would either remove this line or explain that while _typically_ rejections are handled synchronously in JavaScript it is _possible_ to handle them asynchronously by attaching a `.catch` later on (inside a setTimeout) for example.

Reply:
> Promise are also generally not "handled" they are just "reacted to".

Even if "reacted to" is more correct from a spec perspective, I think it's important to consider the audience we're targeting, who might not be familiar with the term "reacted to" (for example, [on the MDN page for Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise), the terminology "reacted to" is never used, but handlers is. Also, using "reacted to" might cause dissonance with "unhandled rejection" in the survey. If others feel strong about this terminology we can change, but I think sticking to "handled" here will be clearer for participants.

I see your point about the general phrasing though. This is a hard topic to phrase :). For example, IMO `_typically_ rejections are handled synchronously` gives the wrong impression that you can react to a rejection on the same turn it happened, when in truth it will only be handled in the future, even if we do `try { async Promise.reject() } catch { }`. I'll try to rephrase it in a way it covers what you're saying.